### PR TITLE
[DNM] Hydrogen Gas

### PR DESCRIPTION
**Pull request by DaedalusGame**
:cl: DaedalusGame, More Robust Than You, Buggy123
add: Added Hydrogen Gas
add: Added Water Vapor Synthesis
/:cl:

- [x] Initial Gas
- [x] Water Vapor Synthesis
- [x] Hydrogen Explosion
- [ ] Cleanup Reaction Code
- [ ] Visuals for Synthesis (Making it looks like fire requires extra changes)
- [x] Canister forme (Sprites needed)
- [ ] Supermatter Interaction (?)
- [ ] Improve Hydrogen Burn formula
- [ ] Tesla Electrolysis
- [x] Hydrogen Reagent dumping out as gas

[Hydrogen Explosion in Action](https://i.imgur.com/bYFm5VF.gifv)

### CODE REVIEW

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -357,13 +361,70 @@
...
+			return NO_REACTION
+
+		cached_results[id] = cached_last_results[id]
+		if(cached_gases[/datum/gas/oxygen][MOLES] > HYDROGEN_DETONATION_THRESHOLD)
+			var/ignition_moles = min(cached_gases[/datum/gas/hydrogen][MOLES],cached_gases[/datum/gas/oxygen][MOLES]*2)
+			cached_results[id] += ignition_moles
+
+			if(cached_results[id] >= 0)
+				if(istype(location))
+					explosion(location,0,0,2,adminlog = FALSE,flame_range = 2) //If we log this oranges will yell at me
```
Comment (nicbn) [inline code/modules/atmospherics/gasmixtures/reactions.dm]: The ultimate lag

Comment (DaedalusGame) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: it works great!

**Comment by Spaceinaba** [comment]
now we just need carbon so we can make propane and atmos techs can become hank hill

**Comment by Buggy123** [comment]
Plasma is already sorta carbon ish? It burns into carbon dioxide, if nothing else.

**Comment by DaedalusGame** [comment]
According to the defines, tritium is the most carbon you can get

**Comment by Tlaltecuhtli** [comment]
how do u get water? heating water to 100K?

**Comment by Tlaltecuhtli** [comment]
water vapour*

**Comment by DaedalusGame** [comment]
You burn hydrogen with oxygen into water vapor

**Comment by arsserpentarium** [comment]
water should be able to be separated to oxygen and hydrogen

**Comment by DaedalusGame** [comment]
Probably, but how? I'd say it'd be possible to add an electrolysis reaction of sorts to something like grounding rods when they get hit with tesla lightning.

**Comment by Buggy123** [comment]
>According to the defines, tritium is the most carbon you can get

What?


W-what?



What?!



# What.

**Comment by arsserpentarium** [comment]
maybe when emitter rays penetrates gas?
also good idea to burn H+CO2-->H2O+welding_fuel_vapour

**Comment by DaedalusGame** [comment]
Dunc said that gas types are expensive on performance so I'll stray from cluttering this pr with another gas, i already had to do minor changes to reactions to realize hydrogen detonation.

Emitter Rays/Particle Accelerator penetrating gas should be left for fusion or similar I think. Electrolysis by radiation sounds sketchy af

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 2]
```diff
@@ -137,6 +137,12 @@
 	gas_type = /datum/gas/water_vapor
 	filled = 1
 
+/obj/machinery/portable_atmospherics/canister/hydrogen
+	name = "hydrogen canister"
+	desc = "Hydrogen. Who would win in a fight, a tetravalent main component of all life or one thicc proton??"
```
Comment (F-OS) [inline code/modules/atmospherics/machinery/portable/canister.dm]: thicc

**Comment by AverageJoe82** [comment]
H+CO2-->H2O+welding_fuel_vapour should be 2H+CO2 -->H2O+CO(carbon monoxide) realistically, but I don't think it would naturally flow in that direction or if it would need activation energy

**Comment by nicbn** [comment]
Hydrogen gas is H2 not H

**Comment by DaedalusGame** [comment]
I love that we found so many people who know so much about a topic i know very little about: molecular organic chemistry

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 1 above)
Comment (nicbn) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: Also I think explosions should scale according to partial pressure of H2+O2, creating hull breaches if possible.

**Comment by nicbn** [comment]
# HYDROGEN IS INORGANIC

**Comment by DaedalusGame** [comment]
but hydrocarbons are organic

**Comment by nicbn** [comment]
Hydro*carbons* are not hydrogen
Basically it is organic only when there is carbon.

**Comment by DaedalusGame** [comment]
i'm convinced that welding fuel vapor would be a hydrocarbon

**Comment by nicbn** [comment]
Yes but hydrogen wouldn't
Please do not add more reactions than the burn in this PR

**Review by nicbn — CHANGES_REQUESTED** [review]
Make the reagent hydrogen spawn hydrogen gas when spilled (1 unit is 1 mol of gas)

**Comment by DaedalusGame** [comment]
I'm adding electrolysis and nobody can stop me from making mad scientist setups involving tesla coils a reality

**Comment by Buggy123** [comment]
So, I was thinking about what burning hydrogen should look like, and then I realized that at any concentration where it could burn, a even mixture of hydrogen/oxygen will always explode. And then when I thought about it more, I realized this applies to pretty much _all_ flammable gasses and flames from bunsen burners, etc are only flames because the two gases aren't pre-mixed; all plasma fires should in reality be plasma _explosions_.

What I'm trying to say is I have no idea what hydrogen fires should look like because they're fictional, so I'm not going to be much help there, sorry.

**Comment by DaedalusGame** [comment]
Aka make it look like normal fire gotcha boss

**Comment by AverageJoe82** [comment]
fuck im actually retarded, thanks @nicbn

**Comment by nicbn** [comment]
Don't add electrolysis
Believe me, hydrogen is too op to be obtainable via reactions
It is odorless, colorless, nontoxic.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -1777,7 +1789,7 @@
 	id = "synthpax"
 	description = "A colorless liquid that suppresses violence on the subjects. Cheaper to synthetize, but wears out faster than normal Pax."
 	metabolization_rate = 1.5 * REAGENTS_METABOLISM
-  
+
```
Comment (DaedalusGame) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: ok i guess

**Comment by DaedalusGame** [comment]
Not allowing an atmos gas to be produced by reaction is an affront to our gods

**Comment by arsserpentarium** [comment]
@AverageJoe82,I know, but it is 2d space game.And welding fuel is already gas.

**Comment by arsserpentarium** [comment]
@DaedalusGame If you add electrolysis, you should add tesla dischargers.

**Comment by DaedalusGame** [comment]
You can use tesla coils, and welding fuel is not a gas

**Comment by duncathan** [comment]
ughh i need to think about this one. still not even convinced i want it yet

**Comment by arsserpentarium** [comment]
1.found it,
2.I forgot, it was on polaris.

**Comment by DaedalusGame** [comment]
that's why it's DNM tagged, Dunc

**Comment by MoreRobustThanYou** [comment]
@duncathan wait for proper SM interactions before judging please

**Comment by DaedalusGame** [comment]
I was thinking something like that, also adding MORE radiation to tritium makes hydrogen which is less radioactive really makes you think

**Comment by arsserpentarium** [comment]
not sure about second.But it could  be fairly interesting scheme, if we will use hydrogen for "homemade" rad collector.and process tritium to energy somehow.

**Comment by optimumtact** [comment]
closing until dunc wakes up from his eternal sleep

**Comment by Buggy123** [comment]
He was arguing with me earlier today dunc wat

**Pull request closed without merging**